### PR TITLE
Rotate acu towards mid

### PR DESCRIPTION
**Pull request by Crotalus**
To make start position more equal, otherwise north has a slight advantage when maps is about walking immediately to mid

### CODE REVIEW

**Comment by ChrisKitching** [comment]
Isn't the rotation something that can be set by the map authors for each start position?

If so, this is probably not something we should override ourselves.

**Comment by speed2CZ** [comment]
No, map authors only place markers with starting positions.

**Comment by Sheeo** [comment]
They're denoted in the save file using army markers which _do_ have an orientation:

```lua
['ARMY_4'] = {
    ['color'] = STRING( 'ff800080' ),
    ['type'] = STRING( 'Blank Marker' ),
    ['prop'] = STRING( '/env/common/props/markers/M_Blank_prop.bp' ),
    ['orientation'] = VECTOR3( 0, -0, 0 ),
    ['position'] = VECTOR3( 330.5, 26, 50.5 ),
},
```

If this orientation isn't respected when placing the commander, we should just make it so?

**Comment by aeoncleanse** [comment]
I agree. This should be done via the map scripts.

**Comment by speed2CZ** [comment]
With that you mean updating every map so we have this cool feature? Idk if thats the best way to do it. Also its not super obvious to rotate the markers in the map editor. That might take ages before people learn to use it.

**Comment by Crotalus** [comment]
No one will ever update old maps so the correct solution is probably to assume 0, 0, 0 as default rotation in which we rotate towards mid, otherwise use the one designated by the map creator.

**Comment by Crotalus** [comment]
Seems like the orientation set in map wasn't respected at all so fixed that + lobby option to rotate the ACU towards mid.

**Comment by speed2CZ** [comment]
Only if you wanted to break your map and it would require editing save.lua with some text editor. Map editor support only rotation around Y axis.

Other thing is I cant imagine game where you would not want ACU rotated. I think its an improvement that should be always enabled

**Comment by speed2CZ** [comment]
I should also mention that you can't type the rotation in the editor to have it same for both players like you can do with units, but only rotate manualy with mouse which is not perfect.

**Comment by ChrisKitching** [comment]
> I should also mention that you can't type the rotation in the editor to have it same for both players like you can do with units, but only rotate manualy with mouse which is not perfect.

Editor bugs should not influence code in not-the-editor.

**Comment by speed2CZ** [comment]
Just that not many ppl will do it if it can't be done easily in the map editor.

**Comment by Sheeo** [comment]
I like this solution.

**Comment by Zockyzock** [comment]
I don't think we want to start inflating lobby options morr, if it's a good idea it can be implemented and be always enabled, but if there are significant downsides the small improvement doesn't warrant a new option.

**Comment by JaggedAppliance** [comment]
Rotation works correctly, only problem is you can start walking before your acu has actually gated in :D

**Comment by aeoncleanse** [comment]
@Crotalus What is the status on this? I think we want it defaulted, not an optional thing.

**Comment by Crotalus** [comment]
Updated the functionality to always be on as default now without a visible lobby option. If a map / mod wants to disable the rotation it's possible to manually specify ScenarioOptions.RotateACU == 'Marker' to make it respect map marker rotation or 'Off' (in reality anything else than 'On' / 'Marker') to disable it.

Also rebased onto latest `develop`

File: lua/sim/ScenarioUtilities.lua [hunk 1]
```diff
@@ -338,7 +337,17 @@ function CreateInitialArmyGroup(strArmy, createCommander)
...
             end
-            cdrUnit:RotateTowardsMid()
+
+            local rotateOpt = ScenarioInfo.Options['RotateACU']
+            if not rotateOpt or rotateOpt == 'On' then
+                cdrUnit:RotateTowardsMid()
+            elseif rotateOpt == 'Marker' then
+                local marker = GetMarker(strArmy) or {}
+                if marker['orientation'] then
+                    local o = EulerToQuaternion(unpack(marker['orientation']))
```
Comment (aeoncleanse) [inline lua/sim/ScenarioUtilities.lua]: What does this function do?

File: lua/sim/Unit.lua [hunk 2]
```diff
@@ -1817,6 +1817,17 @@ Unit = Class(moho.unit_methods) {
         self:SetRotation(angle + current_yaw)
     end,
 
+    RotateTowards = function(self, tpos)
```
Comment (aeoncleanse) [inline lua/sim/Unit.lua]: This function is unused. Do we need it?

File: lua/sim/ScenarioUtilities.lua [hunk 3]
```diff
@@ -338,6 +337,17 @@ function CreateInitialArmyGroup(strArmy, createCommander)
                 cdrUnit:HideBone(0, true)
                 ForkThread(CommanderWarpDelay, cdrUnit, 3)
             end
+
+            local rotateOpt = ScenarioInfo.Options['RotateACU']
+            if not rotateOpt or rotateOpt == 'On' then
+                cdrUnit:RotateTowardsMid()
+            elseif rotateOpt == 'Marker' then
+                local marker = GetMarker(strArmy) or {}
+                if marker['orientation'] then
```
Comment (aeoncleanse) [inline lua/sim/ScenarioUtilities.lua]: To be clear, map-makers would need to set this value in which file, and where in that file?

Comment (Crotalus) [inline lua/sim/ScenarioUtilities.lua]: In the corresponding map files, maps/<mapname>/<mapname>_save.lua. There's a 'orientation' key in the same data structure as the marker position for each start position. Not sure the current map editor supports changing it there (speed2 remark earlier in this PR).

File: lua/sim/ScenarioUtilities.lua (hunk 1 above)
Comment (Crotalus) [inline lua/sim/ScenarioUtilities.lua]: Converts the marker's euler-orientation to the corresponding quaternion values which SetOrientation likes.

https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles

File: lua/sim/Unit.lua (hunk 2 above)
Comment (Crotalus) [inline lua/sim/Unit.lua]: `RotateTowards()` is used by `RotateTowardsMid()`. If you meant `Rotate()` I left that in if we in the future wants to rotate something relative to the current orientation.

**Comment by aeoncleanse** [comment]
d16cdcba69fb297c0bf4ebf4f33a417b0054d160

**Pull request closed without merging**